src/components/ComboBox/ComboBox.tsx

```diff
@@ -202,8 +209,15 @@ const ComboBox = React.forwardRef<HTMLDivElement, ComboBoxProps>(function ComboB
   }, [inputValue, selectedOptionDisplayValue, selectedOption, memoizedGetOptionLabel, options]);
 
   const handleOpen = useCallback(
-    (isOpen: boolean) => {
-      setIsOpen(isOpen);
+    (isOpen: boolean, newValue?: string) => {
+      const hasMetMinimumCharacterCount =
+        (newValue?.length && newValue.length >= countOfCharsBeforeDropdown) ||
+        inputValue.length >=
+          (countOfCharsBeforeDropdown === 1 ? countOfCharsBeforeDropdown : countOfCharsBeforeDropdown - 1);
```

Comment:
This logic was a bit hard to read for me, does this make sense to you as a clearer version? (which has bonus clamping of the new value minimum, partly just because I think parallel aids comprehension)

```suggestion
      const inputValueMinimum  = Math.max(1, countOfCharsBeforeDropdown - 1);
      const newValueMinimum = Math.max(0, countOfCharsBeforeDropdown);

      const hasMetMinimumCharacterCount =
        newValue?.length >= newValueMinimum ||
        inputValue.length >= inputValueMinimum;

```